pkg/expapi/types.go

```diff
@@ -27,3 +27,40 @@ support is experimental.
 */
 
 package expapi
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// SubresourceReference contains enough information to let you inspect or modify the referred subresource.
+type SubresourceReference struct {
+	Kind        string `json:"kind,omitempty"`
+	Namespace   string `json:"namespace,omitempty"`
+	Name        string `json:"name,omitempty"`
+	APIVersion  string `json:"apiVersion,omitempty"`
+	Subresource string `json:"subresource,omitempty"`
+}
+
+// HorizontalPodAutoscaler represents the configuration of a horizontal pod autoscaler.
+type HorizontalPodAutoscaler struct {
+	api.TypeMeta   `json:",inline"`
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Target is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current resource consumption from its status,
+	// and will set the desired number of pods by modyfying its spec.
+	Target *SubresourceReference `json:"target"`
+	// MinCount is the lower limit for the desired number of pods that can be set by the autoscaler.
+	MinCount int `json:"minCount"`
+	// MaxCount is the upper limit for the desired number of pods that can be set by the autoscaler. It cannot be smaller than MinCount.
+	MaxCount int `json:"maxCount"`
+	// TargetCPUConsumption is the target average CPU consumption of pods that the autoscaler will try to maintain by adjusting the desired number of pods.
+	TargetCPUConsumption float64 `json:"targetCPUConsumption"`
```

Comment:
I've changed TargetCPUConsumption into more general resource target. I hope this satisfies your requirements. In the first step, two resource types will be supported: "memory" and "cpu". 

Please note that we want to proceed with the implementation incrementally: start from the simplest version and then extend it and generalize it. That's also why our API is "experimental":  most probably it will be changed (and extended) in future. I hope proposed API is sufficient for the first version.